### PR TITLE
feat: PA-VM Integration with Panorama Server

### PR DESCRIPTION
**Pull request by kpdhulipala**
feat: PA-VM Integration with Panorama Server
Allow additional parameters in vendor config (panoramaAuthKey, panoramaIpAddress)

### CODE REVIEW

File: equinix/data_source_network_device.go [hunk 1]
```diff
@@ -661,6 +661,19 @@ func createDataSourceVendorConfigurationSchema() map[string]*schema.Schema {
 			Sensitive:   true,
 			Description: neDeviceVendorConfigDescriptions["LicenseID"],
 		},
+		neDeviceVendorConfigSchemaNames["PanoramaIPAddress"]: {
+			Type:        schema.TypeString,
+			ForceNew:    true,
```
Comment (rling-equinix) [inline equinix/data_source_network_device.go:666]: We don't need forceNew tag for computed attributes.

File: equinix/data_source_network_device.go [hunk 2]
```diff
@@ -661,6 +661,19 @@ func createDataSourceVendorConfigurationSchema() map[string]*schema.Schema {
...
+			Computed:    true,
+			Description: neDeviceVendorConfigDescriptions["PanoramaIPAddress"],
+		},
+		neDeviceVendorConfigSchemaNames["PanoramaAuthKey"]: {
+			Type:        schema.TypeString,
+			Sensitive:   true,
+			ForceNew:    true,
```
Comment (rling-equinix) [inline equinix/data_source_network_device.go:673]: Same here.

**Review by rling-equinix — CHANGES_REQUESTED** [review]
Are all the doc changes generated automatically?

File: equinix/data_source_network_device.go (hunk 2 above)
Comment (kpdhulipala) [reply equinix/data_source_network_device.go]: addressed

File: equinix/data_source_network_device.go (hunk 1 above)
Comment (kpdhulipala) [reply equinix/data_source_network_device.go]: addressed

**Comment by kpdhulipala** [comment]
> Are all the doc changes generated automatically?

added separate example file to cover PA-VM cluster with panorama specific fields in vendor config

File: examples/resources/network_device/example_9.tf [hunk 3]
```diff
@@ -0,0 +1,45 @@
+# Create PA-VM firewall cluster with Panorama Server Integration
```
Comment (ctreatma) [inline examples/resources/network_device/example_9.tf:1]: This won't automatically be picked up by the docs generator.  You will need to update `templates/resources/network_device.md` to add this example to the template.

To have the new attributes show up in the generated docs, you would have to either add them to the template manually or replace [these lines in the template](https://github.com/equinix/terraform-provider-equinix/blob/main/templates/resources/network_device.md.tmpl#L41-L176) with `{{ .SchemaMarkdown | trimspace }}`.  A similar change would need to be made in `templates/data-sources/network_device.md` in order to automatically generate attributes in the docs for that data source.

Once the templates are updated you will need to run `make docs` and commit the changes.

Comment (kpdhulipala) [reply examples/resources/network_device/example_9.tf:1]: thanks @ctreatma , looks like now these are generated rightly and Go Tests / Validate docs (pull_request)  check is passed.

**Review by rling-equinix — APPROVED** [review]
Lgtm.

**Review by schadalawada — APPROVED** [review]

**Review by ctreatma — APPROVED** [review]
Rubber-stamping since this is all specific to Network Edge and it's already approved by that team.  Will need to follow up separately to adjust codeowners so that DRE isn't tagged for review of PRs like this one.

**Pull request merged**